### PR TITLE
Fix some acceptance tests bugs

### PR DESCRIPTION
**Pull request by ishustava**
Changes proposed in this PR:
* Fix a bug where if the namespace is set in your kube context, the tests will ignore it
* Update client-go, terratest, and consul versions

### CODE REVIEW

File: test/acceptance/framework/consul_cluster.go [hunk 1]
```diff
@@ -103,25 +105,29 @@ func (h *HelmCluster) Destroy(t *testing.T) {
 	helm.Delete(t, h.helmOptions, h.releaseName, false)
 
 	// delete PVCs
-	h.kubernetesClient.CoreV1().PersistentVolumeClaims(h.helmOptions.KubectlOptions.Namespace).DeleteCollection(&metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: "release=" + h.releaseName})
+	h.kubernetesClient.CoreV1().PersistentVolumeClaims(h.helmOptions.KubectlOptions.Namespace).DeleteCollection(context.TODO(), metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: "release=" + h.releaseName})
```
Comment (lkysow) [inline test/acceptance/framework/consul_cluster.go:108]: will we ever be able to pass in a ctx here? If not then we could use .background

**Review by lkysow — APPROVED** [review]

File: test/acceptance/framework/environment.go [hunk 2]
```diff
@@ -70,25 +71,48 @@ type kubernetesContext struct {
...
+		require.NoError(t, err)
+
+		contextName := helpers.KubernetesContextFromOptions(t, k.options)
+		if rawConfig.Contexts[contextName].Namespace != "" {
+			k.options.Namespace = rawConfig.Contexts[contextName].Namespace
+		} else {
+			k.options.Namespace = metav1.NamespaceDefault
+		}
+	}
+	return k.options
```
Comment (ashwin-venkatesh) [inline test/acceptance/framework/environment.go:107]: 🏅

File: test/acceptance/go.mod [hunk 3]
```diff
@@ -3,12 +3,13 @@ module github.com/hashicorp/consul-helm/test/acceptance
 go 1.14
 
 require (
-	github.com/gruntwork-io/terratest v0.27.3
-	github.com/hashicorp/consul/api v1.5.0
-	github.com/hashicorp/consul/sdk v0.5.0
+	github.com/gruntwork-io/terratest v0.29.0
+	github.com/hashicorp/consul/api v1.6.0
+	github.com/hashicorp/consul/sdk v0.6.0
+	github.com/hashicorp/serf v0.9.3
```
Comment (ashwin-venkatesh) [inline test/acceptance/go.mod:9]: What required a dependency on `serf` out of curiosity?

**Review by thisisnotashwin — APPROVED** [review]
Have a question about the addition about the serf dependency but that's about it.

This looks great

File: test/acceptance/framework/consul_cluster.go (hunk 1 above)
Comment (ashwin-venkatesh) [reply test/acceptance/framework/consul_cluster.go:108]: +1 to using `context.Background()` in tests. From the context docs:

```
// Background returns a non-nil, empty Context. It is never canceled, has no
// values, and has no deadline. It is typically used by the main function,
// initialization, and tests, and as the top-level Context for incoming
// requests.
```

File: test/acceptance/go.mod (hunk 3 above)
Comment (ishustava) [reply test/acceptance/go.mod:9]: Good catch! *shakes fist at git rebase* I've rebased this branch with master and this retry keeps haunting me!

The answer is we never need a serf dependency 😄

File: test/acceptance/framework/consul_cluster.go (hunk 1 above)
Comment (ishustava) [reply test/acceptance/framework/consul_cluster.go:108]: I don't think we need to pass ctx at the moment. Let me switch to background instead, thank you both!

File: test/acceptance/go.mod (hunk 3 above)
Comment (ashwin-venkatesh) [reply test/acceptance/go.mod:9]: lol. yeah "rebase reintroducing things i fixed" has bitten me many a time.

**Pull request merged**